Optionally use threads in command_loop

kylebebak opened the pull request:
Closes https://elementary.atlassian.net/browse/ERS-855

- adds `use_procs` arg to `command_loop`, which defaults to true, but can be set to false to use threads for command loop instead of processes
- removes the need to do this kind of stuff, e.g. in integration framework

```py
        command_thread = Thread(
            target=self.element._command_loop,
            args=(
                self.element._command_loop_shutdown,
                0,
            ),
            daemon=True,
        )
```

Dan and I agreed it would be better to go with the multi-element fix to https://elementary.atlassian.net/browse/ERS-851, especially because we want to hotfix this back into 2.3

Still, this is a good feature to add, and once it's verified working we can use this solve for wherever we want a multi-threaded command loop

dpipemazo commented on languages/python/atom/element.py:
:+1: nice. This is a cool way of doing it.

dpipemazo commented on languages/python/atom/element.py:
```diff
@@ -1167,33 +1171,18 @@ def command_loop(
         if n_procs <= 0:
             raise ValueError("n_procs must be a positive integer")
 
-        # note: This warning is emitted in situations where the calling process
```
:+1: Thank you, this warning was a bit confusing and not meeting its intended purpose. I think we'll be OK without it.

aywaldron commented on languages/python/atom/element.py:
```diff
@@ -1138,6 +1139,7 @@ def wait_for_elements_healthy(
     def command_loop(
         self,
         n_procs: int = 1,
+        use_procs: bool = True,
```
I think this API is slightly confusing now since we're using `n_procs` for both processes and threads, while `use_procs` indicates processes and not threads. Perhaps we could change the name of `n_procs` to be more generic and not include "procs"?

dpipemazo replied:
+1 on this, "workers" might be more generic.

aywaldron commented on languages/python/atom/element.py:
```diff
@@ -1154,6 +1156,8 @@ def command_loop(
             n_procs: Number of worker processes.  Each worker process will pull
                 work from the Element's shared command consumer group (defaults
```
Should update docstring to indicate either processes or threads

aywaldron reviewed (commented):
This is great Kyle! Just a comment on the API naming which I think could be improved

kylebebak commented:
Changed name of `n_procs` arg to `n_workers`, and moved `use_procs` arg to end of `command_loop` signature to ensure client code can't confuse `block` and `use_procs` args
